diff --git a/include/configs/rk3328_common.h b/include/configs/rk3328_common.h
index bd2bfe29103..a5e7d5add37 100644
--- a/include/configs/rk3328_common.h
+++ b/include/configs/rk3328_common.h
@@ -21,7 +21,7 @@
 	"fdt_addr_r=0x01e00000\0"	\
 	"fdtoverlay_addr_r=0x01f00000\0"	\
 	"kernel_addr_r=0x02080000\0"	\
-	"ramdisk_addr_r=0x06000000\0"	\
+	"ramdisk_addr_r=0x0a200000\0"	\
 	"kernel_comp_addr_r=0x08000000\0"	\
 	"kernel_comp_size=0x2000000\0"
 
